.NET/Microsoft.Recognizers.Text.DateTime/Parsers/BaseDateTimePeriodParser.cs

```diff
@@ -138,7 +230,7 @@ private DateTimeResolutionResult ParseSimpleCases(string text, DateObject refere
                 }
 
                 // parse following date
-                var er = this.Config.DateExtractor.Extract(trimedText.Substring(match.Length));
+                var er = this.Config.DateExtractor.Extract(trimedText.Replace(match.Value, ""));
```

Comment:
What's the desired behaviour here? Just reading the code it's hard to understand if there's any side effect.